### PR TITLE
Issue #76. Create custom embed to display GeoCollect missions data

### PR DESCRIPTION
**Pull request by mricca**
This is the config used to test.
```
{
	"version": 2,
	"map": {
		"center": {
		    "x": 8.790401458740227,
            "y": 44.4239167027002,
            "crs": "EPSG:4326"
        },
        "maxExtent": [-20037508.34, -20037508.34, 20037508.34, 20037508.34],
        "projection": "EPSG:900913",
        "units": "m",
        "zoom": 13,
        "layers": [{
                "id": "mapnik__0",
                "group": "background",
                "source": "osm",
                "name": "mapnik",
                "title": "Open Street Map",
                "type": "osm",
                "visibility": true,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false
            }, {
                "id": "SATELLITE__1",
                "group": "background",
                "source": "google",
                "name": "SATELLITE",
                "title": "Google Satellite",
                "type": "google",
                "visibility": false,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false
            }, {
                "id": "BASE_CARTOGRAFICA__2",
                "group": "background",
                "name": "BASE_CARTOGRAFICA",
                "title": "BASE_CARTOGRAFICA",
                "type": "wms",
                "url": "/geoserver-test/wms",
                "bbox": {
                    "crs": "EPSG:4326",
                    "bounds": {
                        "minx": "8.664631331583704",
                        "miny": "44.378093234171516",
                        "maxx": "9.096015724443996",
                        "maxy": "44.52038356628868"
                    }
                },
                "visibility": false,
                "singleTile": false,
                "allowedSRS": {
                    "EPSG:3003": true,
                    "EPSG:3857": true,
                    "EPSG:4326": true,
                    "EPSG:900913": true,
                    "CRS:84": true
                },
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "params": {}
            }, {
                "id": "OpenTopoMap__3",
                "group": "background",
                "source": "OpenTopoMap",
                "name": "OpenTopoMap",
                "provider": "OpenTopoMap",
                "title": "OpenTopoMap",
                "type": "tileprovider",
                "visibility": false,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false
            }, {
                "id": "rv:rv1__4",
                "format": "image/jpeg",
                "group": "background",
                "name": "rv:rv1",
                "title": "e-Geos Ortofoto RealVista 1.0",
                "type": "wms",
                "url": "http://www.realvista.it/reflector/open/service",
                "visibility": false,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "params": {}
            }, {
                "id": "undefined__5",
                "group": "background",
                "source": "ol",
                "title": "Nessuno sfondo",
                "type": "OpenLayers.Layer",
                "visibility": false,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false
            }, {
                "id": "muro__6",
                "search": {
                    "url": "http://geocollect.geo-solutions.it:80/geoserver/it.geosolutions/wfs?",
                    "type": "wfs"
                },
                "name": "muro",
                "title": "Muri",
                "type": "wms",
                "url": "http://geocollect.geo-solutions.it/geoserver/it.geosolutions/ows",
                "bbox": {
                    "crs": "EPSG:4326",
                    "bounds": {
                        "minx": "8.0",
                        "miny": "44.0",
                        "maxx": "10.0",
                        "maxy": "45.0"
                    }
                },
                "visibility": false,
                "singleTile": false,
                "allowedSRS": {
                    "EPSG:3003": true,
                    "EPSG:4326": true,
                    "EPSG:900913": true,
                    "CRS:84": true
                },
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "params": {}
            }, {
                "id": "cens_muri_sop__7",
                "search": {
                    "url": "http://geocollect.geo-solutions.it:80/geoserver/it.geosolutions/wfs?",
                    "type": "wfs"
                },
                "name": "cens_muri_sop",
                "title": "Sopralluoghi",
                "type": "wms",
                "url": "http://geocollect.geo-solutions.it/geoserver/it.geosolutions/ows",
                "bbox": {
                    "crs": "EPSG:4326",
                    "bounds": {
                        "minx": "8.0",
                        "miny": "43.0",
                        "maxx": "12.0",
                        "maxy": "45.0"
                    }
                },
                "visibility": true,
                "singleTile": false,
                "allowedSRS": {
                    "EPSG:3003": true,
                    "EPSG:4326": true,
                    "EPSG:900913": true,
                    "CRS:84": true
                },
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "featureInfo": {
                    "format": "JSON",
                    "viewer": {
                        "type": "Geocollect"
                    }
                },
                "params": {}
            }, {
                "id": "cens_muri__8",
                "search": {
                    "url": "http://geocollect.geo-solutions.it:80/geoserver/it.geosolutions/wfs?",
                    "type": "wfs"
                },
                "name": "cens_muri",
                "opacity": 1,
                "title": "Segnalazioni Muri",
                "type": "wms",
                "url": "http://geocollect.geo-solutions.it/geoserver/it.geosolutions/ows",
                "bbox": {
                    "crs": "EPSG:4326",
                    "bounds": {
                        "minx": "8.0",
                        "miny": "44.0",
                        "maxx": "11.0",
                        "maxy": "45.0"
                    }
                },
                "visibility": true,
                "singleTile": false,
                "allowedSRS": {
                    "EPSG:3003": true,
                    "EPSG:4326": true,
                    "EPSG:900913": true,
                    "CRS:84": true
                },
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "featureInfo": {
                    "format": "JSON",
                    "viewer": {
                        "type": "Geocollect"
                    }
                },
                "params": {}
            }
        ],
        "groups": [{
                "id": "Default",
                "expanded": true
            }
        ]
    },
    "catalogServices": {
        "services": {
            "GeoPortale Comune di Genova": {
                "url": "/geoserver-test/csw",
                "type": "csw",
                "title": "GeoPortale Comune di Genova",
                "autoload": true
            },
            "GeoCollect": {
                "url": "http://geocollect.geo-solutions.it/geoserver/it.geosolutions/ows",
                "type": "wms",
                "title": "GeoCollect",
                "autoload": false
            }
        }
    }
}

```

### CODE REVIEW

File: js/enhancers/geocollectViewerEnhancer.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        baseUrl = baseUrl + '?action=get_filelist&folder=' + folder + "/" + response.features[0].id;
+        return axios.get(baseUrl);
+    }).map((data) => (
+        {imgsResponse: data}
+    ));
+
+};
+
+const twoCall = (response) => {
+    const url = 'http://geocollect.geo-solutions.it/geoserver/it.geosolutions/ows';
```
Comment (offtherailz) [inline js/enhancers/geocollectViewerEnhancer.jsx]: maybe this should be passed from the layer's viewer config

File: js/enhancers/geocollectViewerEnhancer.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    const gcid = response.features[0].properties.gcid;
+    const myOrigId = response.features[0].properties.my_orig_id;
+
+    if (myOrigId) {
+        let baseUrl = "http://geocollect.geo-solutions.it/opensdi2-manager/mvc/fileManager/extJSbrowser";
+        const folder = "/media/cens_muri/" + response.features[0].properties[featureProperty];
```
Comment (offtherailz) [inline js/enhancers/geocollectViewerEnhancer.jsx]: same as above

File: assets/css/custom.css [hunk 3]
```diff
@@ -50,3 +50,25 @@
     background-repeat: no-repeat;
     background-size: contain;
 }
+
+div.geocollect-sop .panel-title {
```
Comment (offtherailz) [inline assets/css/custom.css]: I see this style do not work in embedded view

File: js/viewer/GeocollectViewer.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        if (wfsResponse) {
+            if (wfsResponse.features.length > 0) {
+                return (
+                        <Tab eventKey={2} key={'Modifica'} title={'Modifiche'} >
+                            <Accordion defaultActiveKey={0}>
+                            {
+                                (wfsResponse.features || []).map((feature, i) => {
+                                    return (
+                                        <Panel className="geocollect-sop" key={i} header={'Modifica del ' + moment(feature.properties.gc_created).format("DD-MM-YYYY, h:mm:ss a")} eventKey={i} style={{position: "relative"}}>
+                                            <RowViewer key={feature.properties.gc_created} exclude={["bbox"]} {...feature.properties}/>
```
Comment (offtherailz) [inline js/viewer/GeocollectViewer.jsx]: Missing images for each "Modifica"

**Comment by mricca** [comment]
How do you think about this?

video_2017-11-02_152618

**Comment by mricca** [comment]
What do you think about this?

video_2017-11-02_152618

**Pull request merged**